libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -55,6 +69,7 @@ export class BookSearchComponent implements OnInit, OnDestroy {
 
   addBookToReadingList(book: Book) {
     this.store.dispatch(addToReadingList({ book }));
+    this.openSnackbar('Added to Reading List', 'Undo',book);
```

Comment:
Here, you are opening the snackbar the moment the user adds a book to the reading list. How do you know that the book was successfully added, to allow it to be removed (undo)? You should make sure that the `addToReadingList` was successful, in order to allow its undoing, or else you could run into errors! You could try a different place to open the snackbar from, where all these actions are available! (hint: effects)